### PR TITLE
feat(IncognitoTyping): add incognito typing extension

### LINKED ISSUE #387: feat(IncognitoTyping): Extension to add the ability to toggle on and off emiting typing status.
**Description:** Create a new extension called `IncognitoTyping` which will allow you to click an icon next to the chatbar in order to disable sending our typing status to the remote user. This should be toggle-able. 

**Requirements:**

- [ ] Create extension named `IncognitoTyping` under `extensions/native`.
- [ ] Should render icon next to the chatbar.
- [ ] Should be able to click the icon to toggle hiding typing on and off.
- [ ] Should have tooltip (`ui_kit::Tooltip`) popover that says "Hide my typing".

**Assets:**
Included in the our Icon repo is now a `Shape::Keyboard` and `Shape::KeyboardSlash`. Use these for the icon.

| View      | Resource |
| ----------- | ----------- |
| Default      | image       |
| Enabled   | image        |

**Figma Mock:**
https://www.figma.com/file/5XfbetOXEEkawpmYcEWvwb/Typing-Indicator-Extension?node-id=0%3A1&t=yQT1t9R91zKnDqie-0

### PR DESCRIPTION
**Pull request by pavlzk**
### What this PR does 📖
- add incognito typing extension
- update state to make it accessible by extensions

### Which issue(s) this PR fixes 🔨
- Resolve #387

### CODE REVIEW

**Comment by InfamousVague** [comment]
This looks good to me, I think it'd be worth trying to see if we can make this one into a dylib and load it as a proper extension instead of importing it in the future. It's okay if we want to split this into a new issue though.

File: src/components/reusable/textarea.rs [hunk 1]
```diff
@@ -93,10 +94,12 @@ pub fn TextArea<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
                     return;
                 }
             };
-            chan2.send(ChanCmd::Typing {
-                chat_id,
-                rg: cx.props.messaging.clone(),
-            });
+            if send_typing {
```
Comment (dariusc93) [inline src/components/reusable/textarea.rs:97]: If the extension is disabled or not available for whatever reason wouldnt this always be false?

Comment (pavlzk) [reply src/components/reusable/textarea.rs:97]: We could set is `true` as default and add this option to app settings where it can also be changed.

**Review by dariusc93 — APPROVED** [review]
From a glance it looks good.

**Review by phillsatellite — APPROVED** [review]

**Pull request merged**